client/client.go

```diff
@@ -46,21 +59,23 @@ type Currency string
 // CurrencyUSD is the type of the currency of USC
 const CurrencyUSD = "USD"
 
-// OfferPayload is the term Payload of the price
-type OfferPayload struct {
+// ChargePayload is the charge payload of the offer
+type ChargePayload struct {
 	LeaseContractLength string `json:"leaseContractLength"`
 	PurchaseOption      string `json:"purchaseOption"`
 }
 
 // Offer is the api message of an Offer
 type Offer struct {
-	ID         string
-	InstanceID string
+	ID string
+
+	OfferType  OfferType
+	InstanceID string // If OfferType is not Instance, InstanceID would be nil
 
-	Type OfferType
-	// Term is nil when the ChargeType is onDemand
-	Payload *OfferPayload
+	ChargeType    ChargeType
+	ChargePayload *ChargePayload // Payload is nil when the ChargeType is onDemand
```

Comment:
nit: typically comments are added before the field with public comment. Also fix other places.

```
// ChargePayload is nill when ...
ChargePayload *ChargePayload
```